app/src/androidTest/java/org/mozilla/fenix/ui/robots/DownloadRobot.kt

```diff
@@ -133,16 +133,36 @@ fun downloadRobot(interact: DownloadRobot.() -> Unit): DownloadRobot.Transition
 }
 
 private fun assertDownloadPrompt(fileName: String) {
-    assertTrue(
-        "Download prompt button not visible",
-        mDevice.findObject(UiSelector().resourceId("$packageName:id/download_button"))
-            .waitForExists(waitingTime)
-    )
-    assertTrue(
-        "$fileName title doesn't match",
-        mDevice.findObject(UiSelector().text(fileName))
-            .waitForExists(waitingTime)
-    )
+    var currentTries = 0
+    while (currentTries++ < 3) {
+        try {
+            assertTrue(
+                "Download prompt button not visible",
+                mDevice.findObject(UiSelector().resourceId("$packageName:id/download_button"))
+                    .waitForExists(waitingTime)
+            )
+            assertTrue(
+                "$fileName title doesn't match",
+                mDevice.findObject(UiSelector().text(fileName))
+                    .waitForExists(waitingTime)
+            )
```

Comment:
missing a `break` statement, so if we don't error we don't need to iterate again, similar to [this](https://github.com/mozilla-mobile/fenix/blob/f9369a5535180526c0621230a26e57dcb8cc4f65/app/src/androidTest/java/org/mozilla/fenix/ui/robots/CustomTabRobot.kt#L90)

Reply:
Yes, overlooked it, done